### DISCUSSION TITLE
Why moon is not using moon?

### DISCUSSION THREAD

**Discussion by artsiommiksiuk** [moonrepo/moon#1585] [General]
I run through the repo and see that moon tasks are defined only on workspace level (so all are inherited) and only for js. Most of the CI is done via shell scripts, and root is using justfile defined commands to build rust and whole project.

There aren't much examples in the web around moon usage and when even moon not showing "how to" that's quite a bit discouraging.

**Comment by milesj** [comment]
It's rather simple. To run moon in debug mode, I have to use `cargo run` to compile the current Rust code. So to emulate something like `moon check --all` would be `cargo run -- check --all`. Putting Rust/Cargo into moon's configuration + tasks adds a completely unnecessary layer, as it would be `cargo run -- run project:task`, which in turn runs the moon task `project:task`, which would then spawn another `cargo` child process that would basically just be `cargo run` again.

That's why moon only uses moon for our JS packages. Using Rust in moon provides no benefits. If anything it's drawbacks.
